src/main_benchmark.cpp

```diff
@@ -343,48 +452,110 @@ int main(int argc, char** argv) {
     ///////////////////////
     // setup workload machinery
     ///////////////////////
+    std::vector<OVMS_InferenceRequest*> requests;
     std::vector<std::unique_ptr<std::thread>> workerThreads;
     std::vector<std::promise<void>> startSignals(threadCount);
-    std::vector<std::promise<void>> stopSignals(threadCount);
+    std::vector<std::promise<void>> readySignals(threadCount);
     std::vector<std::future<void>> futureStartSignals;
-    std::vector<std::future<void>> futureStopSignals;
+    std::vector<std::future<void>> futureReadySignals;
     std::vector<size_t> wholeThreadsTimesUs(threadCount, 0);
     std::transform(startSignals.begin(),
         startSignals.end(),
         std::back_inserter(futureStartSignals),
         [](auto& p) { return p.get_future(); });
-    std::transform(stopSignals.begin(),
-        stopSignals.end(),
-        std::back_inserter(futureStopSignals),
+    std::transform(readySignals.begin(),
+        readySignals.end(),
+        std::back_inserter(futureReadySignals),
         [](auto& p) { return p.get_future(); });
 
     ///////////////////////
     // prepare threads
     ///////////////////////
-    for (size_t i = 0; i < threadCount; ++i) {
-        workerThreads.emplace_back(std::make_unique<std::thread>(
-            [&futureStartSignals,
-             &futureStopSignals,
-             &niterPerThread,
-             &wholeThreadsTimesUs,
-             &wholeTimes,
-             &pureTimes,
-             &srv,
-             &request,
-             i]() {
-                triggerInferenceInALoop(
-                    futureStartSignals[i],
-                    futureStopSignals[i],
-                    niterPerThread,
-                    wholeThreadsTimesUs[i],
-                    wholeTimes[i],
-                    pureTimes[i],
-                    srv,
-                    request);
-            }));
+    if (mode == Mode::INFERENCE_ONLY) {
+        for (size_t i = 0; i < threadCount; ++i) {
+            workerThreads.emplace_back(std::make_unique<std::thread>(
+                [&futureStartSignals,
+                &readySignals,
+                &niterPerThread,
+                &wholeThreadsTimesUs,
+                &wholeTimes,
+                &pureTimes,
+                &srv,
+                &request,
+                i]() {
+                    triggerInferenceInALoopInferenceOnly(
```

Comment:
To make the code shorter, you could pick a function pointer depending on mode before line 474. (assuming all 3 functions have the same parameters). Then, you could call `workerThreads.emplace_back` only once with previously picked function pointer (the same behavior can be observed in our unit tests)